specification/eventgrid/data-plane/Microsoft.Devices/stable/2018-01-01/IotHub.json

```diff
@@ -0,0 +1,150 @@
+      "properties": {
+        "authenticationType": {
+          "type": "string",
+          "description": "Authentication type used for this device: either SAS, SelfSigned, or CertificateAuthority."
+        },
+        "cloudToDeviceMessageCount": {
+          "type": "number",
+          "description": "Count of cloud to device messages sent to this device."
+        },
+        "connectionState": {
+          "type": "string",
+          "description": "Whether the device is connected or disconnected."
+        },
+        "deviceId": {
+          "type": "string",
+          "description": "The unique identifier of the device twin."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A piece of information that describes the content of the device twin. Each etag is guaranteed to be unique per device twin."
+        },
+        "lastActivityTime": {
+          "type": "string",
+          "description": "The ISO8601 timestamp of the last activity."
+        },
+        "properties": {
+          "type": "object",
+          "description": "Properties JSON element.",
+          "properties": {
+            "desired": {
+              "description": "A portion of the properties that can be written only by the application back-end, and read by the device.",
+              "$ref" : "#/definitions/DeviceTwinProperties"
+            },
+            "reported": {
+              "description": "A portion of the properties that can be written only by the device, and read by the application back-end.",
+              "$ref" : "#/definitions/DeviceTwinProperties"
+            }
+          }
+        },
+        "status": {
+          "type": "string",
+          "description": "Whether the device twin is enabled or disabled."
+        },
+        "statusUpdateTime": {
+          "type": "string",
+          "description": "The ISO8601 timestamp of the last device twin status update."
+        },
+        "version": {
+          "type": "number",
+          "description": "An integer that is incremented by one each time the device twin is updated."
+        },
+        "x509Thumbprint": {
+          "type": "object",
+          "description": "The thumbprint is a unique value for the x509 certificate, commonly used to find a particular certificate in a certificate store. The thumbprint is dynamically generated using the SHA1 algorithm, and does not physically exist in the certificate.",
+          "properties": {
+            "primaryThumbprint": {
+              "type": "string",
+              "description": "Primary thumbprint for the x509 certificate."
+            },
+            "secondaryThumbprint": {
+              "type": "string",
+              "description": "Secondary thumbprint for the x509 certificate."
+            }
+          }
+        }
+      }
+    },
+    "DeviceTwinProperties": {
```

Comment:
nit: Use a more meaningful name here, there already exists a definition for `DeviceTwinInfo` with some properties
something like `TwinDeviceReportedProperties` or `DeviceReportedProperties` based on the property name

Reply:
in this context reported properties has a different meaning. So using that would only confuse the reader who understands the iot hub twin concepts. DeviceTwinProperty is the right name.

Also, TwinInfo and TwinProperties were names I picked from another swagger doc that a partner team of us already uses. I would like to be consistent with those. 

I will pass on this nit.